include/dd/Package.hpp

```diff
@@ -2890,6 +2890,266 @@ template <class Config> class Package {
+  size 2^k. For each part, keeps the leftmost column unchanged, and sets the
+  remaining columns to 0.
+  **/
+  mEdge setColumnsToZero(const mEdge& e, Qubit k) {
+    if (e.isTerminal()) {
+      return e;
+    }
+    if (k == 0) {
+      return e;
+    }
+    mEdge eCopy{e.p, Complex::one()};
+    // check if it's in the compute table with an edge weight of one
+    if (const auto* r = setMatrixColumnsToZero.lookup(eCopy, k); r != nullptr) {
+      auto f = *r;
+      f.w = cn.lookup(e.w * f.w);
+      return f;
+    }
+    // the matrix of the current DD has dimensions 2^h x 2^h
+    const auto h = e.p->v + 1;
+    std::array<mEdge, NEDGE> edges{};
+    edges[0] = setColumnsToZero(e.p->e[0], k);
+    edges[2] = setColumnsToZero(e.p->e[2], k);
+    if (k < h) {
+      edges[1] = setColumnsToZero(e.p->e[1], k);
+      edges[3] = setColumnsToZero(e.p->e[3], k);
+    } else {
+      edges[1] = mEdge::zero();
+      edges[3] = mEdge::zero();
+    }
+    auto f = makeDDNode(e.p->v, edges);
+    // add to the compute table with a weight of 1
+    setMatrixColumnsToZero.insert(eCopy, k, f);
+    f.w = cn.lookup(e.w * f.w);
+    return f;
+  }
+
+  /**
+      Equally divides the rows of the matrix represented by e intro parts of
+  size 2^k. For each part, keeps the top row unchanged, and sets the remaining
+  entries to 0.
+  **/
+  mEdge setRowsToZero(const mEdge& e, Qubit k) {
+    if (e.isTerminal()) {
+      return e;
+    }
+    if (k == 0) {
+      return e;
+    }
+    mEdge eCopy{e.p, Complex::one()};
+    // check if it's in the compute table with an edge weight of one
+    if (const auto* r = setMatrixRowsToZero.lookup(eCopy, k); r != nullptr) {
+      auto f = *r;
+      f.w = cn.lookup(e.w * f.w);
+      return f;
+    }
+    // the matrix of the current DD has dimensions 2^h x 2^h
+    const auto h = e.p->v + 1;
+    std::array<mEdge, NEDGE> edges{};
+    edges[0] = setRowsToZero(e.p->e[0], k);
+    edges[1] = setRowsToZero(e.p->e[1], k);
+    if (k < h) {
+      edges[2] = setRowsToZero(e.p->e[2], k);
+      edges[3] = setRowsToZero(e.p->e[3], k);
+    } else {
+      edges[2] = mEdge::zero();
+      edges[3] = mEdge::zero();
+    }
+    auto f = makeDDNode(e.p->v, edges);
+    // add to the compute table with a weight of 1
+    setMatrixRowsToZero.insert(eCopy, k, f);
+    f.w = cn.lookup(e.w * f.w);
+    return f;
+  }
+
+private:
+  mEdge partialEquivalenceCheckSubroutine(mEdge& u, Qubit m, Qubit k,
+                                          Qubit extra) {
+    auto u1{u};
+    // add extra ancillary qubits
+    if (extra > 0) {
+      if (u.p->v + 1U + extra > nqubits) {
+        resize(u.p->v + 1U + extra);
+      }
+      auto idExtra = makeIdent(extra);
+      u1 = kronecker(u, idExtra);
+    }
+    auto u2 = setColumnsToZero(u1, k);
+    auto u3 = shiftAllRows(u2, m);
+    auto u4 = multiply(conjugateTranspose(u1), u3);
+    auto u5 = setRowsToZero(u4, k);
```

Comment:
just wondering: you pass `k` to both of the methods that set something to zero. In the paper, they pass `d` to those methods. Any particular reason for the difference here?